src/Interpreters/JIT/CHJIT.cpp

```diff
@@ -483,15 +493,43 @@ std::string CHJIT::getMangledName(const std::string & name_to_mangle) const
     return mangled_name;
 }
 
-void CHJIT::runOptimizationPassesOnModule(llvm::Module & module) const
+#ifndef NDEBUG
+
+static std::string DumpModuleIR(const llvm::Module& module)
 {
+    std::string ir;
+    llvm::raw_string_ostream stream(ir);
+    module.print(stream, nullptr);
+    return ir;
+}
+
+#endif
+
+void CHJIT::runOptimizationPassesOnModule(llvm::Module & module, llvm::TargetIRAnalysis target_analysis) const
```

Comment:
You don't need to change the method signature to pass `TargetIRAnalysis`. It's possible to extract it right here.